static/app/views/dashboards/datasetConfig/metrics.tsx

```diff
@@ -393,13 +393,13 @@ function getMetricRequest(
   const requestData = getMetricsApiRequestQuery(
     {
       field: query.aggregates[0],
-      query: query.conditions,
-      groupBy: query.columns,
-      orderBy: query.orderby,
+      query: query.conditions || undefined,
+      groupBy: query.columns || undefined,
+      orderBy: query.orderby || undefined,
     },
     pageFilters,
     {
-      limit,
+      limit: limit || undefined,
```

Comment:
@iambriccardo  This will also strip `limit = 0`, though I think there is no real world use-case for it.